es67x/src/main/scala/tech/beshu/ror/es/utils/ThreadRepo.scala

```diff
@@ -20,11 +20,19 @@ import org.elasticsearch.rest.RestRequest
 import tech.beshu.ror.accesscontrol.domain.UriPath
 import tech.beshu.ror.es.RorRestChannel
 
+import scala.util.{Failure, Success, Try}
+
 object ThreadRepo {
   private val threadLocalChannel = new ThreadLocal[RorRestChannel]
 
-  def setRestChannel(restChannel: RorRestChannel): Unit = {
+  def safeSetRestChannel(restChannel: RorRestChannel)(code: => Unit): Unit = {
```

Comment:
Something seems wrong across ES-specific modules about closing/removing the channel
- the channel is set here
- it is closed here in this method in case of exception
- otherwise, it is closed in the `RorRestChannel#sendResponse` method

But in modules for ES 6.7 and all 7.0 except 716 and 717, it seems that channel is never removed. Is that correct?

Maybe this method can be renamed to `def withRestChannel` and it could handle both setting and removing the channel? Then it would be unnecessary to remove channel in `RorRestChannel#sendResponse`.

Reply:
ok, this is fixed. Now, we clean the restChannel in every module.
Thanks for noticing it.